src/audio/src/tune/sof_src_lite_int32.m

```diff
@@ -0,0 +1,37 @@
+%% Export full set of conversions for IPC4
+
+% SPDX-License-Identifier: BSD-3-Clause
+%
+% Copyright(c) 2022 Intel Corporation. All rights reserved.
+
+fs1 = [ 8 11.025 12 16 18.9 22.05 24 32 37.8 44.1 48 64 88.2 96 176.4 192 ] * 1e3;
+
+fs2 = [8 16 24 32 44.1 48 88.2 96 176.4 192] * 1e3;

```

Comment:
This is not looking right, in the earlier conversions matrix there was only 32000, 44100, 48000 as fs1 and 16000, 48000 as fs2. See https://github.com/thesofproject/sof/blob/88366121b3f50a4fe86675860af9426b8b946240/src/audio/src/coef/src_lite_ipc4_int32_table.h#L28

You won't get correct rates vs. indices with this this big table. You should remove columns and rows for all other rates.

Reply:
Huh, this file didn't save correctly it seems, or git messed something up.  I'll take a look.

Reply:
Should be good now

Reply:
Can you check if you still need the C code change for same rates check

Reply:
The .h files were generated correctly, I just copied the wrong .m file. So the result should be the same.

Reply:
Yes you are right, your fix is correct. The src_in_fs[] and src_out_fs[] table is not correct. When I play with src_lite benchmark topology 32 kHz to 48 kHz I get these traces:

[  453.788055] <inf> src: src_polyphase_init: idx_in = 0, idx_out = 1
[  453.788063] <inf> src: src_polyphase_init: src_in_fs = 8000, src_out_fs = 11025
[  453.788070] <inf> src: src_polyphase_init: in_fs = 32000, out_fs = 48000

There's apparently race with other global arrays and your way avoids it. I think there is need for one more fix but you can leave it to me.

Reply:
Nice, thanks. I pushed fixes for the .m file now.

Reply:
@lyakh @tobonex I'm worried about how the src_in_fs[] and src_out_fs[] tables from big SRC have appeared to SRC-lite context. If that table somehow lurks, can we trust the other important structures like src_stage are not messed up? If src_stage is messed up the FIR coeffcients will be wrong and conversion quality will be not what it should be.

In src_lite_int32_table.h

const int src_in_fs[3] = { 32000, 44100, 48000};
const int src_out_fs[2] = { 16000, 48000};

and in src_ipc4_int32_table.h

static const int src_in_fs[16] = { 8000, 11025, 12000, 16000, 18900, 22050, 24000, 32000,
	 37800, 44100, 48000, 64000, 88200, 96000, 176400, 192000
	};
static const int src_out_fs[10] = { 8000, 16000, 24000, 32000, 44100, 48000, 88200, 96000,
	 176400, 192000};

So, the proposed code fix is a workaround to a larger issue.